scripts/installer.sh

```diff
@@ -0,0 +1,251 @@
+#!/bin sh
+# Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+#
+# This script detects the current operating system, and installs
+# Tailscale according to that OS's conventions.
+
+set -eu
+
+# All the code is wrapped in a main function that gets called at the
+# bottom of the file, so that a truncated partial download doesn't end
+# up executing half a script.
+main() {
+	# Step 1: detect the current linux distro, version, and packaging system.
+	#
+	# We rely on /etc/os-release to get OS information. With some
+	# distros, we have to normalize the information we get into the
+	# form that we need to drive our install process.
+
+	OS=
+	VERSION=
+	PACKAGETYPE=
+
+	if [ ! -f /etc/os-release ]; then
+		echo "/etc/os-release not found, cannot get OS information."
+		echo "Please contact us at support@tailscale.com for install help."
```

Comment:
"for help installing" or "for installation help" or "for help" ?

Reply:
Rather than contact us, it'd be best to link them to a page where they can choose things for themselves. Maybe this is [tailscale.com/download](https://tailscale.com/download), but [our Linux installation docs](https://tailscale.com/kb/1031/install-linux/) is certainly a stable link too.

Reply:
We reworked this logic in teamflow. There's now a single error path, which triggers only when we've already fallen off the end of all our known, supported systems. In that scenario, there's not really much we can point people at, other than "tell us what kind of OS this was, so we can figure out if support is plausible".